--- icu-4.4.ebuild
+++ icu-4.4.ebuild
@@ -4,7 +4,7 @@
 
 EAPI="3"
 
-inherit flag-o-matic versionator
+inherit flag-o-matic versionator autotools
 
 DESCRIPTION="International Components for Unicode"
 HOMEPAGE="http://www.icu-project.org/ http://ibm.com/software/globalization/icu/"
@@ -40,12 +40,14 @@
 }
 
 src_prepare() {
+	epatch "${FILESDIR}"/${P}-ctestfw-makefilein.patch
 	# Do not hardcode used CFLAGS, LDFLAGS etc. into icu-config
 	# Bug 202059
 	# https://bugs.icu-project.org/trac/ticket/6102
 	for x in ARFLAGS CFLAGS CPPFLAGS CXXFLAGS FFLAGS LDFLAGS; do
 		sed -i -e "/^${x} =.*/s:@${x}@::" "config/Makefile.inc.in" || die "sed failed"
 	done
+	eautomake
 }
 
 src_configure() {
