--- vboxdrv/Makefile.orig	2009-04-13 23:20:42.000000000 +0800
+++ vboxdrv/Makefile	2009-04-13 23:21:11.000000000 +0800
@@ -66,6 +66,9 @@
  $(warning Using BUILD_TYPE='$(BUILD_TYPE)' from the $(origin BUILD_TYPE).)
 endif
 
+# use vm_insert_page() API (if available) to map kernel pages to userland - better debugging 
+VBOX_USE_INSERT_PAGE = 1
+
 # override is required by the Debian guys
 override MODULE = vboxdrv
 OBJS   = \
